src/tools/utils/editorOperationUtils.ts

```diff
@@ -0,0 +1,114 @@
+}
+
+function splitOperationText (
+  model: monaco.editor.ITextModel,
+  uneditableRanges: monaco.Range[],
+  text: string | null
+): (string | null)[] {
+  if (text == null || text === '') {
+    return uneditableRanges.length > 0 ? [] : [text]
+  }
+
+  const splitText: string[] = []
+  const uneditableRangesText = uneditableRanges.map(range => model.getValueInRange(range))
+  let currentRange: number = 0
+  let textToSplit: string = text
+  while (textToSplit !== '' && currentRange < uneditableRangesText.length) {
+    const rangeText = uneditableRangesText[currentRange]
+    if (rangeText != null && rangeText !== '') {
+      const rangeTextIndex = textToSplit.indexOf(rangeText)
+
+      if (rangeTextIndex === -1) {
+        return []
+      }
+
+      const currentUneditableRange = uneditableRanges[currentRange]!
+      if (rangeTextIndex !== 0) {
+        let textToKeep = textToSplit.slice(0, rangeTextIndex)
+        if (textToKeep.endsWith('\n') && currentUneditableRange.startColumn === 1) {
+          textToKeep = textToKeep.slice(0, textToKeep.length - 1)
+        }
+        splitText.push(textToKeep)
+      }
+
+      const uneditableRangeMaxEndColumn = model.getLineMaxColumn(currentUneditableRange.endLineNumber)
+      textToSplit = textToSplit.slice(rangeTextIndex + rangeText.length)
+      if (textToSplit.startsWith('\n') && currentUneditableRange.endColumn === uneditableRangeMaxEndColumn) {
+        textToSplit = textToSplit.slice(1, textToSplit.length)
+      }
+    }
+    currentRange++
+  }
+
+  if (textToSplit !== '') {
+    splitText.push(textToSplit.endsWith('\n') ? textToSplit.slice(0, textToSplit.length - 1) : textToSplit)
+  }
+  return splitText
+}
+
+export function splitOperationsForLockedCode (
+  model: monaco.editor.ITextModel,
+  operations: ValidAnnotatedEditOperation[],
+  uneditableRanges: monaco.Range[]
+): ValidAnnotatedEditOperation[] {
+  let newOperations: ValidAnnotatedEditOperation[] = []
+  for (const operation of operations) {
+    const {
+      firstRanges: operationEditableRanges,
+      secondRanges: operationUneditableRanges
+    } = minusRanges(model, operation.range, uneditableRanges)
+    const splitText = splitOperationText(model, operationUneditableRanges, operation.text)
+    newOperations = [
+      ...newOperations,
+      ...createNewOperationsFromRanges(operation, operationEditableRanges, splitText)
+    ]
+  }
+  return newOperations
+}
+
+export function computeNewOperationsForLockedCode (
+  model: monaco.editor.ITextModel,
+  decorationFilter: (decoration: monaco.editor.IModelDecoration) => boolean,
+  editorOperations: ValidAnnotatedEditOperation[],
+  withDecoration: boolean
+): ValidAnnotatedEditOperation[] {
+  const uneditableRanges = getLockedRanges(model, decorationFilter, withDecoration)
+  if (uneditableRanges.length <= 0) {
+    return editorOperations
+  }
+
+  return splitOperationsForLockedCode(model, editorOperations, uneditableRanges)
```

Comment:
this function name should probably be similar to the called function name

Reply:
[refactor: rename functions on locked code and their variables](https://github.com/CodinGame/monaco-editor-wrapper/pull/125/commits/4211c36f490bd0ebf871eeb479fac4afd597cb2d)